ducktape/utils/wait.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2015 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Conscious waiting
+#
+#
+#
+
+import time
+import traceback
+import os
+import errno
+import signal
+
+class TimeoutError (Exception):
+    pass
+
+time_saved_sum = 0
+
+def time_saved ():
+    """ Returns the amount of time saved using wait.until() instead of time.sleep() """
+    global time_saved_sum
+    return time_saved_sum
+
+
+def sleep (seconds):
+    """ Simple wrapper for time.sleep() but indicates that the programmer
+        made a conscious decisions not to use wait.until() """
+    return time.sleep(seconds)
+
+def until0 (timeout, expect_result, test_func, *args):
```

Comment:
This is potentially useful externally and thus deserves a snazzier name - perhaps `until_result`?
Also, why not include `**kwargs` in each of the three existing `until*` methods?